### ISSUE TITLE
resolved DNSSEC=yes gets fixated on broken servers -> no DNS for you

### ISSUE THREAD

**Issue by FauxFaux** [systemd/systemd#7040]
### Submission type

 - Bug report

### systemd version the issue has been seen with

> 234-2ubuntu12

### Used distribution

> Ubuntu Artful/17.10 prerelease

### Unexpected behaviour you saw

I have configured `resolved` to use Google Public DNS, but my distro packaging helpfully ensures it also sees the useless resolver my ISP's CPE ("modem router") gives out:

```
% egrep -v '^$|^#' /etc/systemd/resolved.conf
[Resolve]
DNS=8.8.4.4
DNSSEC=yes
```

```
% systemd-resolve --status
Global
         DNS Servers: 8.8.4.4
                      192.168.1.254
          DNS Domain: home
...
```

Initially, this setup works. However, after a few minutes, all DNS queries start failing, apparently forever:

```
% systemd-resolve google.co.uk
google.co.uk: resolve call failed: DNSSEC validation failed: incompatible-server
```

`resolved` is sending the queries to the useless DNS server, and is unhappy that it won't talk DNSSEC.

```
% journalctl -u systemd-resolved | cut ..
Not validating response for 62093, used server feature level does not support DNSSEC.
DNSSEC validation failed for question google.co.uk IN A: incompatible-server
Transaction 62093 for <google.co.uk IN A> on scope dns on */* now complete with <dnssec-failed> from network (unsigned).
Freeing transaction 40464.
Sent message type=error sender=n/a destination=:1.2147 object=n/a interface=n/a member=n/a cookie=25 reply_cookie=2 error=DNSSEC validation failed: incompatible-server
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=26 reply_cookie=0 error=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.2141 object=n/a interface=n/a member=n/a cookie=20 reply_cookie=26 error=n/a
Freeing transaction 62093.
```

Annoyingly that doesn't log that it's using the useless (`192..`) DNS server, but TCP dump sees it:

```
% tcpdump -i em1 'port 53' -An
20:28:49.579605 IP 192.168.1.235.34081 > 192.168.1.254.53: 46893+% A? google.co.uk. (30)
E..:5.@.@............!.5.&.q.-...........google.co.uk.....
20:28:49.579815 IP 192.168.1.235.42331 > 192.168.1.254.53: 16163+% AAAA? google.co.uk. (30)
E..:5.@.@............[.5.&.q?#...........google.co.uk.....
20:28:49.585855 IP 192.168.1.254.53 > 192.168.1.235.34081: 46893 1/0/0 A 216.58.204.67 (46)
E..J....@..i.........5.!.6C..-...........google.co.uk..................:.C
20:28:49.587552 IP 192.168.1.254.53 > 192.168.1.235.42331: 16163 1/0/0 AAAA 2a00:1450:4009:80f::2003 (58)
E..V....@..].........5.[.B..?#...........google.co.uk..............<..*..P@	........ .
```

### Expected behaviour

I would hope that the server would be blacklisted, as it is not useful for resolving things with `DNSSEC=yes`. At worst, I would expect only 50% of queries to fail, instead of 100% of queries forever.

### Steps to reproduce the problem

I presume this would be reproducible `systemd-resolved` configured to talk to one competent DNS server, and one useless one, but I do not know how or where to find a public, useless DNS server, nor how to host one.

(Also: the github editor goes a bit nuts if you keep getting NXDOMAINs served while you're editing a report, I advise against it.)

**Comment by poettering** [comment]
Fix waiting in #7591.

**Comment by paulmenzel** [comment]
Hmm, Heroku DNS (or their users) is currently broken, and the problem happened again.

**Comment by toreanderson** [comment]
This issue is a duplicate of #6490, which will be fixed by pull #17535.
